### ISSUE TITLE
[A11Y] Action Menu contains unnecessary tabstops

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#1968]
## Issue Details

The Action Menu has incorrect focus order when it contains interactive elements (such as an ActionMenuLinkItem).

### Short Description

[2.4.3](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#navigation-mechanisms-focus-order) Focus Order: If a [Web page](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html#webpagedef) can be [navigated sequentially](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html#nav-seqdef) and the navigation sequences affect meaning or operation, focusable components receive focus in an order that preserves meaning and operability. (Level A)

### Impact

- [ ] Blocker
- [ ] Critical
- [x] Serious
- [ ] Moderate
- [ ] Minor

### Steps To Reproduce:

1. Open https://cauldron.dequelabs.com/components/ActionMenu
2. Navigate to "Actions with Links" button
3. Press <kbd>↓</kbd> or open the menu with <kbd>enter</kbd>
4. First item appears focused

### Current Behavior:

When pressing <kbd>tab</kbd> once the menu is open that contains link items, each link item presents itself as an additional tabstop while focus is also managed via `aria-activedescendent`.

### Expected Behavior:

Focus should only be managed via `aria-activedescendent` and there should be no additional tabstops inside of the menu.

---

### Auditor ID / URL

N/A

### Additional Details

<details>
<summary>Expand for extra context</summary>
- [ ] Discovered during VPAT
- [ ] Found using NVDA
- [x] Found using Chrome Screen Reader
- [ ] Found with special configuration 
</details>

---

## Screenshots

![](https://github.com/user-attachments/assets/6a79f99b-3970-4162-9bff-916f8470dd63)

### 1. Perceivable

<details> 
<summary>Expand</summary>

#### 1.1 Text Alternatives

- [ ] 1.1.1 Non-text Content

#### 1.2 Time-based Media

- [ ] 1.2.1 Audio-only and Video-only (Prerecorded)
- [ ] 1.2.2 Captions (Prerecorded)
- [ ] 1.2.3 Audio Description or Media Alternative (Prerecorded)
- [ ] 1.2.4 Captions (Live)
- [ ] 1.2.5 Audio Description (Prerecorded)

#### 1.3 Info and Relationships

- [ ] 1.3.1 Info and Relationships
- [ ] 1.3.2 Meaningful Sequence
- [ ] 1.3.3 Sensory Characteristics
- [ ] 1.3.4 Orientation
- [ ] 1.3.5 Identify Input Purpose

#### 1.4 Distinguishable

- [ ] 1.4.1 Use of Color
- [ ] 1.4.2 Audio Control
- [ ] 1.4.3 Contrast (Minimum)
- [ ] 1.4.4 Resize text
- [ ] 1.4.5 Images of Text
- [ ] 1.4.10 Reflow
- [ ] 1.4.11 Non-text Contrast
- [ ] 1.4.12 Text Spacing
- [ ] 1.4.13 Content on Hover or Focus
</details>

### 2. Operable

<details> 
<summary>Expand</summary>

#### 2.1 Keyboard Accessible

- [ ] 2.1.1 Keyboard
- [ ] 2.1.2 No Keyboard Trap
- [ ] 2.1.4 Character Key Shortcuts

#### 2.2 Enough Time

- [ ] 2.2.1 Timing Adjustable
- [ ] 2.2.2 Pause, Stop, Hide

#### 2.3 Seizures and Physical Reactions

- [ ] 2.3.1 Three Flashes or Below Threshold

#### 2.4 Navigable

- [ ] 2.4.1 Bypass Blocks
- [ ] 2.4.2 Page Titled
- [x] 2.4.3 Focus Order
- [ ] 2.4.4 Link Purpose (In Context)
- [ ] 2.4.5 Multiple Ways
- [ ] 2.4.6 Headings and Labels
- [ ] 2.4.7 Focus Visible

#### 2.5 Input Modalities

- [ ] 2.5.1 Pointer Gestures
- [ ] 2.5.2 Pointer Cancellation
- [ ] 2.5.3 Label in Name
- [ ] 2.5.4 Motion Actuation
</details>

### 3. Understandable

<details> 
<summary>Expand</summary>

#### 3.1 Readable

- [ ] 3.1.1 Language of Page
- [ ] 3.1.2 Language of Parts

#### 3.2 Predictable

- [ ] 3.2.1 On Focus
- [ ] 3.2.2 On Input
- [ ] 3.2.3 Consistent Navigation
- [ ] 3.2.4 Consistent Identification

#### 3.3 Input Assistance

- [ ] 3.3.1 Error Identification
- [ ] 3.3.2 Labels or Instructions
- [ ] 3.3.3 Error Suggestion
- [ ] 3.3.4 Error Prevention (Legal, Financial, Data)
</details>

### 4. Robust

<details>
<summary>Expand</summary>

#### 4.1 Compatible

- [ ] 4.1.1 Parsing
- [ ] 4.1.2 Name, Role, Value
- [ ] 4.1.3 Status Messages
</details>
